cmd/relay-gossip/main.go

```diff
@@ -99,46 +100,157 @@ var runCmd = &cli.Command{
 			}
 		}
 
-		c, err := lib.Create(cctx, cctx.IsSet(metricsFlag.Name))
+		var hashMaps []string
+		if cctx.IsSet(lib.HashListFlag.Name) {
+			hashMaps = cctx.StringSlice(lib.HashListFlag.Name)
+		}
+
+		hashFlagSet := cctx.IsSet(lib.HashFlag.Name)
+		if hashFlagSet {
+			if cctx.IsSet(lib.HashListFlag.Name) {
+				return fmt.Errorf("--%s is exclusive with --%s. Use one or the other flag", lib.HashFlag.Name, lib.HashListFlag.Name)
+			}
+
+			hashMaps = append(hashMaps, cctx.String(lib.HashFlag.Name))
+		}
+
+		groupConfs := computeGroupConfs(cctx)
+
+		hashesMap, err := computeHashesMap(hashMaps, groupConfs)
 		if err != nil {
-			return xerrors.Errorf("constructing client: %w", err)
+			return err
 		}
 
-		chainHash := cctx.String(lib.HashFlag.Name)
-		if chainHash == "" {
-			info, err := c.Info(context.Background())
-			if err != nil {
-				return xerrors.Errorf("getting chain info: %w", err)
+		if len(groupConfs) >= 1 &&
+			len(hashesMap) == 0 {
+
+			for _, groupConf := range groupConfs {
+				err = cctx.Set(lib.GroupConfFlag.Name, groupConf)
+				if err != nil {
+					return fmt.Errorf("setting group-conf %q got: %w", groupConf, err)
+				}
+
+				err := boostrapGossipRelayNode(cctx, "")
+				if err != nil {
+					return err
+				}
 			}
-			chainHash = hex.EncodeToString(info.Hash())
 		}
 
-		cfg := &lp2p.GossipRelayConfig{
-			ChainHash:    chainHash,
-			PeerWith:     cctx.StringSlice(peerWithFlag.Name),
-			Addr:         cctx.String(listenFlag.Name),
-			DataDir:      cctx.String(storeFlag.Name),
-			IdentityPath: cctx.String(idFlag.Name),
-			Client:       c,
+		for chainHash, groupConf := range hashesMap {
+			if err := cctx.Set(lib.HashFlag.Name, chainHash); err != nil {
+				return fmt.Errorf("setting client hash: %w", err)
+			}
+
+			if groupConf != "" {
+				err = cctx.Set(lib.GroupConfFlag.Name, groupConf)
+				if err != nil {
+					return fmt.Errorf("setting group-conf for hash %q got: %w", chainHash, err)
+				}
+			}
+
+			if err := boostrapGossipRelayNode(cctx, chainHash); err != nil {
+				return err
+			}
 		}
-		if _, err := lp2p.NewGossipRelayNode(log.DefaultLogger(), cfg); err != nil {
-			return err
+
+		// Try and build a default gossip relay using other options that might have been set.
+		if len(hashesMap) == 0 &&
+			len(groupConfs) == 0 &&
+			!hashFlagSet {
+			if err := boostrapGossipRelayNode(cctx, ""); err != nil {
+				return err
+			}
 		}
-		<-chan int(nil)
-		return nil
+
+		// Wait until we are signaled to shutdown.
+		<-cctx.Context.Done()
+
+		return cctx.Context.Err()
 	},
 }
 
+func computeGroupConfs(cctx *cli.Context) []string {
```

Comment:
perhaps it'd be better to provide a `--group-file-list` param for consistency - comma separated for one flag and multivariate for another could be confusing for operators. Additionally, an operator should pass either group files or hashes, but not both